### PR TITLE
SBOM/maven dev dependency change 17 july

### PR DESCRIPTION
**Pull request by karthika-g**

### CODE REVIEW

File: src/LCT.PackageIdentifier/MavenProcessor.cs [hunk 1]
```diff
@@ -67,6 +84,100 @@ public Bom ParsePackageFile(CommonAppSettings appSettings)
...
+
+        }
+
+        private static List<Component> DevdependencyIdentification(List<Component> ListOfComponents, List<Component> iterateBOM, List<Component> checkBOM)
+        {
+            foreach (var item in iterateBOM)
+            {
+                if (checkBOM.Count == 0)
+                {
+                    SetPropertiesforBOM(ref ListOfComponents, item, "false");
```
Comment (sumanthkb44) [inline src/LCT.PackageIdentifier/MavenProcessor.cs:104]: You can write Comments for what's happening here.

File: src/LCT.PackageIdentifier/MavenProcessor.cs [hunk 2]
```diff
@@ -67,6 +84,100 @@ public Bom ParsePackageFile(CommonAppSettings appSettings)
...
+                {
+                    if (componentList.Exists(x => x.Name == entry.Name))
+                    {
+                        SetPropertiesforBOM(ref componentsToBOM, entry, "false");
+                    }
+                    else
+                    {
+                        SetPropertiesforBOM(ref componentsToBOM, entry, "true");
+
+                        BomCreator.bomKpiData.DevDependentComponents++;
```
Comment (sumanthkb44) [inline src/LCT.PackageIdentifier/MavenProcessor.cs:149]: Check the Code Complexity for this method, maybe it's above 15

File: src/LCT.PackageIdentifier/MavenProcessor.cs (hunk 1 above)
Comment (karthika-g) [reply src/LCT.PackageIdentifier/MavenProcessor.cs:106]: added

File: src/LCT.PackageIdentifier/MavenProcessor.cs (hunk 2 above)
Comment (karthika-g) [reply src/LCT.PackageIdentifier/MavenProcessor.cs:151]: checked it is 12

**Pull request merged**